### PR TITLE
SWITCHYARD-3012 - cloning the Fuse tooling Help files as a start

### PR DESCRIPTION
**Pull request by bfitzpat**

### CODE REVIEW

File: eclipse/plugins/org.switchyard.tools.help/.project [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>org.switchyard.tools.help</name>
+	<comment>This is a shared POM parent for FuseSource Maven projects. NO_M2ECLIPSE_SUPPORT: Project files created with the maven-eclipse-plugin are not supported in M2Eclipse.</comment>
```
Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.help/.project]: maybe FuseSource shoudl be removed?

File: eclipse/plugins/org.switchyard.tools.help/OSGI-INF/l10n/bundle.properties [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#Properties file for org.fusesource.ide.help
+Bundle-Vendor = JBoss by Red Hat
+Bundle-Name = Fuse Plugins for Eclipse Help Plugin
```
Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.help/OSGI-INF/l10n/bundle.properties]: SwitchYard Plugins?

File: eclipse/plugins/org.switchyard.tools.help/OSGI-INF/l10n/bundle.properties [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#Properties file for org.fusesource.ide.help
```
Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.help/OSGI-INF/l10n/bundle.properties]: org.fusesource.ide.help --> org.switchyard.tools.help

File: targetplatform/build.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+[INFO] Scanning for projects...
```
Comment (apupier) [inline targetplatform/build.txt]: I guess this file should not be commited

File: eclipse/plugins/org.switchyard.tools.help/pom.xml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+       	 <groupId>org.apache.maven.plugins</groupId>
+         <artifactId>maven-antrun-plugin</artifactId>
+         <executions>
+           <execution>
+             <phase>package</phase>
+             <configuration>
+               <target>
+                 <!-- copy the generated plugin -->
+                 <copy todir="${basedir}" failonerror="false" overwrite="true">
+                   <fileset dir="${basedir}/../../rider-docs/output/eclipsehelp" excludes="META-INF/**" />
```
Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.help/pom.xml]: is it something still used? there is no more "rider" folders as far as I know

File: eclipse/plugins/org.switchyard.tools.help/index.html [hunk 6]
```diff
@@ -0,0 +1,45 @@
+<html>
+   <head>
+      <meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">
+         <title>Red Hat JBoss Swithyard Tooling for Eclipse</title>
```
Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.help/index.html]: typo Swithyard
and shouldn't it be writtent a cap for Yard? SwitchYard?

File: eclipse/plugins/org.switchyard.tools.help/index.html [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+         <link rel="stylesheet" type="text/css" href="eclipse_book.css">
+            <meta name="generator" content="DocBook XSL Stylesheets V1.77.1">
+            <meta name="keywords" content="ESB, Open Source, open source, integration, OSGi">
+         <link rel="home" href="index.html" title="Red Hat JBoss Switchyard Tooling for Eclipse">
```
Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.help/index.html]: shouldn't it be writtent a cap for Yard? SwitchYard?

File: eclipse/plugins/org.switchyard.tools.help/index.html [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+         <link rel="next" href="chap-JBoss_Integration_and_SOA_Development.html" title="JBoss Integration and SOA Development">
+         <link rel="copyright" href="tmdisclaim.html" title="Trademark Disclaimer">
+   </head>
+   <body bgcolor="white" text="black" link="#0000FF" vlink="#840084" alink="#0000FF">
+      <div class="set" lang="en">
+         <div class="titlepage">
+            <div>
+               <div>
+                  <h1 class="title"><a name="SYEclipseHelp"></a>
+                     Red Hat JBoss Switchyard Tooling for Eclipse
```
Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.help/index.html]: shouldn't it be writtent a cap for Yard? SwitchYard?

**Comment by apupier** [comment]
I think that the build failure is due to the fact that the plugins/pom.xml has not declared the new plugin module org.switchyard.tools.helpl

File: eclipse/plugins/org.switchyard.tools.help/pom.xml (hunk 5 above)
Comment (lhein) [inline eclipse/plugins/org.switchyard.tools.help/pom.xml]: that looks outdated. if thats something we have in fuseide too we should get rid of it

File: eclipse/features/org.switchyard.tools.feature/feature.xml [hunk 9]
```diff
@@ -4,9 +4,9 @@
       label="%featureName"
       version="2.4.0.qualifier"
       provider-name="%featureProvider"
+      plugin="org.switchyard.tools"
```
Comment (lhein) [inline eclipse/features/org.switchyard.tools.feature/feature.xml]: do u have a plugin with that id?

**Comment by bfitzpat** [comment]
Testing-wise it works great locally in the Help system, but thanks for pointing out these issues. I'll take a crack at them.

File: eclipse/plugins/org.switchyard.tools.help/pom.xml (hunk 5 above)
Comment (bfitzpat) [inline eclipse/plugins/org.switchyard.tools.help/pom.xml]: any suggestions on what we should do with that? maybe just change to /output/eclipsehelp ?

File: eclipse/features/org.switchyard.tools.feature/feature.xml (hunk 9 above)
Comment (bfitzpat) [inline eclipse/features/org.switchyard.tools.feature/feature.xml]: Yup

File: eclipse/plugins/org.switchyard.tools.help/pom.xml (hunk 5 above)
Comment (lhein) [inline eclipse/plugins/org.switchyard.tools.help/pom.xml]: when thinking about it that could be something done by the docs team. We should ask Fintan and/or Jane.

Comment (bfitzpat) [inline eclipse/plugins/org.switchyard.tools.help/pom.xml]: Sent an e-mail to the docs team with the question.

**Comment by bfitzpat** [comment]
Addressed all comments (thanks for the feedback and pointing out some of those issues in hidden files I missed) and sent an e-mail to the Docs team about the pom question.

File: eclipse/plugins/org.switchyard.tools.help/pom.xml (hunk 5 above)
Comment (fbolton) [inline eclipse/plugins/org.switchyard.tools.help/pom.xml]: I'm not too sure about this  either, as this POM is not part of our docs workflow. All I know is that 'rider' was an alternative brand for Apache Camel back in the FuseSource days (>5 years ago). So any instance of it sounds old to me.

Comment (bfitzpat) [inline eclipse/plugins/org.switchyard.tools.help/pom.xml]: @fbolton Any suggestions on what to do with this? Do we remove it? Do we still need the antrun?

Comment (fbolton) [inline eclipse/plugins/org.switchyard.tools.help/pom.xml]: My best guess is that this whole antrun thing is obsolete and can be removed. Just delete it and keep your fingers crossed!

**Comment by lhein** [comment]
Please consult Jane. I know that docs team used to have some ant stuff
running. Just to be on the safe side.

Am 26.06.2017 6:24 nachm. schrieb "Fintan Bolton" <notifications@github.com

**Comment by bfitzpat** [comment]
Went ahead and removed the antrun build bits of the pom.xml for now. Everything seems to build ok locally. We can add them back in if we need to after hearing from Jane

**Comment by bfitzpat** [comment]
According to Jane, we should listen to Fintan as the "master of doc Ant build tools." :) So hopefully this will build and then we'll get it merged after an approval from Lars or Aurelien. 👍

**Pull request merged**